version: '3'
services:
  reverse-proxy:
          image: traefik
          command:
              - "--api.insecure=true"
              - "--providers.docker.endpoint=unix:///var/run/docker.sock"  
              - "--providers.docker=true"
              - "--api.insecure=true"
              - "--providers.docker.exposedbydefault=false"
              - "--entrypoints.web.address=:80"
          ports:
                # The HTTP port
                - "80:80"
          volumes:
          # So that Traefik can listen to the Docker events
          - /var/run/docker.sock:/var/run/docker.sock
          labels:
              - "traefik.enable=True"
              - "traefik.http.routers.dashboard.rule=Host(`0.0.0.0`)"
              - "treafik.http.routers.dashboard.service=api@internal"
              - "traefik.http.services.dashboard.loadbalancer.server.port=8080"
              - "traefik.port=8080"
              - "traefik.http.routers.dashboard.middlewares=traefik-auth"
              - "traefik.http.middlewares.traefik-auth.basicauth.users=admin:$$apr1$$8EVjn/nj$$GiLUZqcbueTFeD23SuB6x0"
  database:
    image: postgres:alpine
    env_file:
      - .env/postgres.env
    volumes:
      - ./postgres-data:/var/lib/postgresql/data
    healthcheck:
            test: [ "CMD", "pg_isready", "-q", "-d", "${DB_NAME}", "-U", "postgres" ]
            timeout: 45s
            interval: 10s
            retries: 10
  django:
    image: smvsite/smvsite
    env_file:
      - .env/django.env
    depends_on:
            database:
                    condition:  service_healthy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.django.rule=Host(`127.0.0.1`) && PathPrefix(`/`)"
      - "traefik.http.services.django.loadbalancer.server.port=8000"
      - "traefik.http.routers.traefik.entrypoints=web"
      - "traefik.port=8000"
